app/screens/home/channel_list/categories_list/categories/categories.tsx

```diff
@@ -49,17 +49,25 @@ const Categories = ({
     const isTablet = useIsTablet();
     const switchingTeam = useTeamSwitch();
     const teamId = categories[0]?.teamId;
+    const [showAllCategory, setShowAllCategory] = useState(onlyUnreads);
```

Comment:
There is no need for state. You can get this information from the props already, and they are already causing the re-render. This can be simplified to something like...

```suggestion
    const showAllCategory = onlyUnreads && !unreadsOnTop;
```

Reply:
@larkox This won't work when `onlyUnreads ` & `unreadsOnTop ` is `true`, if both of them are `true`, then we have to make `onlyUnreads` false, so it can render category  `Flatlist`

Reply:
But if both `onlyUnreads` and `unreadsOnTop` are true, this new variable `showAllCategory` is false, and will do the trick, right?

Reply:
@larkox Got it, with this condition
`const showAllCategory = onlyUnreads && !unreadsOnTop;` we don't require any state, 
but one condition needs to be check if `onlyUnreads` and `unreadsOnTop` is true then we have to return `orderedCategories`, so it won't render empty state